tools/c7n_azure/c7n_azure/output.py

```diff
@@ -20,15 +20,29 @@
 import os
 import shutil
 import tempfile
+import uuid
 
 from c7n_azure.storage_utils import StorageUtilities
-from c7n.output import DirectoryOutput, blob_outputs
-
+from c7n_azure.utils import AppInsightsHelper
+from c7n.output import (
+    blob_outputs,
+    log_outputs,
+    metrics_outputs,
+    DirectoryOutput,
+    LogOutput,
+    Metrics
+)
 from c7n.utils import local_session
 
+from applicationinsights import TelemetryClient
+from applicationinsights.logging import LoggingHandler
 from azure.common import AzureHttpError
 
 
+# executionId is used to group logs\metrics from the same run together
+executionId = str(uuid.uuid4())
```

Comment:
execution context (ctx) has an execution id on it already that should be used, and scopes to a single policy execution, which is preferable to do doing a process level global.